site/source/tutorials/1040_dojo1_dijits/index.md

```diff
@@ -0,0 +1,191 @@
+
+{% task 'Wrap Dijits.' %}
+
+{% aside 'Information' %}
+All Dojo 1 modules are AMD modules without default exports.  As such, you need to import Dojo 1 modules using the `import * as Dijit from 'module/id';` syntax.
+{% endaside %}
+
+We recommend any wrapped Dijits be created as individual modules in `/src/dijit` in the same path structure they are located in the source `dijit` package.
+
+{% instruction 'Create `/src/dijit/Fieldset.ts` with the following content:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/dijit/Fieldset.ts' %}
+
+{% instruction 'Create `/src/dijit/form/Button.ts` with the following content:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/dijit/form/Button.ts' %}
+
+{% instruction 'Create `/src/dijit/form/TextBox.ts` with the following content:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/dijit/form/TextBox.ts' %}
+
+{% task 'Import Dijits.' %}
+
+Now that you have wrapped Dijits for use, we need to import them into their containing Dojo 2 or modules.
+
+{% instruction 'Import Dijits into `/src/widgets/WorkerForm.ts`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:5-7 %}
+
+{% section %}
+
+## Placing Dijits
+
+{% task 'Add Dijits to a widget\'s render function.' %}
+
+Now that we have imported our wrapped Dijits into the module they will be used, we need to add them to the render function.
+
+{% instruction 'Add imported Dijits to the `WorkerForm` widget\'s render function:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:44-74 %}
+
+{% aside 'Information' %}
+The <em>properties</em> for a wrapped Dijit are equivalent to the <em>params</em> that can be passed when constructing a Dijit.  For more information on what parameters are possible on Dijits, you can refer to the [Dojo 1 API documentation](https://dojotoolkit.org/api/).
+{% endaside %}
+
+Once Dijits are wrapped, they behave very similar to Dojo 2 widgets.  Coupled with the `dojo-typings` TypeScript typings, you get code completion as well when specifying the _properties_ and TypeScript should help ensure the validity of property values.
+
+One restriction to keep in mind is that wrapped Dijits may only have other wrapped Dijit's as their children. Dojo 2 widgets represent their [DOM virtually](../comingsoon.html), while Dojo 2 Dijits directly manage their own DOM.  Because of this, Dijits cannot properly manage virtual DOM based children.
+
+{% task 'Set a Dojo 1 theme.' %}
+
+When we configured the _externals_ for our project, we included the main CSS for our theme.  Most Dijits will not render properly without a theme being applied to them.  There are many different ways to apply a Dojo 1 theme, but the most straightforward is to set the theme class name on the `<body>` of the application document.
+
+{% instruction 'Add theme class name to `src/index.html`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/index.html' lines:6-8 lang:html %}
```

Comment:
We never import the flat theme's style sheet. Is this missing, or automatically handled?

Reply:
It is implied in the externals... I will be more explicit about it.

Reply:
Actually I was as explicit as I think I can be when discussing the externals, but I will add an aside here to remind people of the _magic_ the build system does.